@geodata/geodata.m

```diff
@@ -556,9 +558,14 @@
                     clear x y demz
                 else
                     % main interpolant
-                    obj.Fb   = griddedInterpolant({x,y},demz,...
-                        'linear','nearest');
-                    obj.x0y0 = [x(1),y(1)];
+                    if obj.BACKUPdemfile~=0

```

Comment:
I find this section quite confusing and I think we should re-write. Why did you put the code to build the secomdary backup interpolant in the logic section that is activated when there is no backup demfile?

Reply:
yes, I was confused as to why we need to use obj.x0y0 to be the x(1) y(1) from the DEM. Can not just always use the bbox lower left-hand corner? bbox is updated to be the DEM extents when none is given already.

Reply:
the code should be in the section for the main dem, not the backup dem. I think the way it is shown on that image is confusing. Perhaps look at the actual code.

Reply:
I did look at the code. There's two sections: one for fromation of the backup interpolant and one for the main interpolant. You placed the backup formation in the section for the main interpolant, which doesn't make sense.

Reply:
that is not true. in the case of backupdem being present then the interpolant for the main dem is set to have no extrapolation so we can detect that nans means outside of the dem extents.

Reply:
Ah I see. yes I was mistaken. In the main interpolation formation, you just turn off extrapolation if a backup is present. Then the back up interpolant is formed after the main DEM is finished. In the mesh size functions, you detect for NaN's and replace it with the secondary one.